### PR TITLE
feat: directly process CRDs

### PR DESCRIPTION
**Pull request by Duologic**
The k3s-setup has proven that processing CRDs into Jsonnet libraries is useful. However
processing them by loading them into a k3s instance has been cumbersome, error prone and
mostly very very slow.

In this PR I've made an attempt at processing CRDs directly in the generator, mitigating
the need for a k3s instance.

This is WIP, TODO for what I know:

- [x] Implement CRDLoader in main.go, reading multiple CRD URLs into the same definition list.
- [x] Rip out the k3s setup.
- [x] Ensure CI can render all! (and measure time, hopefully much faster)
- [ ] Verify the diff with k3s-setup is non-existent.

### CODE REVIEW

**Review by davidovich — COMMENTED** [review]
Note that `swagger` is a trademarked name (by Smartbear), it should probably not be used and the name OpenAPI favored, WDYT ?

File: pkg/swagger/schema.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package swagger
```
Comment (davidovich) [inline pkg/swagger/schema.go:1]: Should this be delegated to something like https://github.com/getkin/kin-openapi for parsing and managing OpenAPI documents ?

Comment (Duologic) [reply pkg/swagger/schema.go:1]: I considered that, even wrote a POC with kin-openapi (nice lib!) this morning. Eventually decided to focus on CRD parsing with the apiextensions.k8s.io itself as I can marshal CRDs directly into structs with that.

The original swagger/openapiv2 parsing could probably use a refactor too, I left the swagger name as it is a bit out of scope for this change right now. Thanks for pointing out that it is trademarked, more the reason to refactor this part.

**Comment by Duologic** [comment]
> Verify the diff with k3s-setup is non-existent.

The diff currently isn't non-existent, but it might not need to be. There are a few changes. Most common is the more specific typing of arguments correctly (T.any->T.<typed>) in docsonnet, but that doesn't change how the jsonnet work, it does introduce a few moare *Mixin() functions as these are now `object` or `array` typed instead of `any`.

Other changes I need to reinspect the diff (which is printed in the logs of the CI job now).

**Comment by Duologic** [comment]
Taking out of 'draft' as it could use a rough review.

File: pkg/swagger/crd.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		return nil, err
+	}
+
+	definitions := []apiextensionsv1.CustomResourceDefinition{}
+	for _, crdfile := range crds {
+		crd := apiextensionsv1.CustomResourceDefinition{}
+
+		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(crdfile), 4096)
+		if err := decoder.Decode(&crd); err != nil {
+			panic(err)
```
Comment (julienduchesne) [inline pkg/swagger/crd.go:37]: panic? Returning an error should do right?
```suggestion
			return nil, err
```

**Review by julienduchesne — COMMENTED** [review]
Looks awesome! I have zero major concerns

File: pkg/swagger/crd.go (hunk 2 above)
Comment (Duologic) [reply pkg/swagger/crd.go:37]: Haha, a leftover from my private POC, nice catch.

File: pkg/swagger/objectmeta.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
+{
```
[the remark this answers is not in the document]
Comment (Duologic) [reply pkg/swagger/objectmeta.json:1]: I manually cut it out of the Kubernetes swagger json, I'll procure a script to automate the update to a version equal to the go modules for k8s.

[the remark this answers is not in the document]
Comment (Duologic) [reply pkg/swagger/objectmeta.json:1]: `make update_objectmeta` now creates this file.

**Review by julienduchesne — APPROVED** [review]
I had a quick glance at the diff. Didn't see anything that would break. As far as I'm concerned, this LGTM! Very nice work!

**Pull request merged**